eth/protocols/snap/sync.go

```diff
@@ -620,6 +620,7 @@ func (s *Syncer) Sync(root common.Hash, cancel chan struct{}) error {
 		trienodeHealResps    = make(chan *trienodeHealResponse)
 		bytecodeHealResps    = make(chan *bytecodeHealResponse)
 	)
+	defer s.commit(true)
```

Comment:
We can put it after Line 586, https://github.com/ethereum/go-ethereum/pull/25681/files#diff-d26ea400bc58a59ffc652f18f3e402fe7e3279c1569c8374c216b5d702f2fe7cR586

It's cleaner for all persistence defer logics.